cluster-autoscaler/core/placeholderpod/create.go

```diff
@@ -0,0 +1,68 @@
+package placeholderpod
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// CreateReplicaSets create one or more replica set(s) of placeholder pods
+func CreateReplicaSets(sets ...ReplicaSet) []*apiv1.Pod {
+	pods := []*apiv1.Pod{}
+	for _, set := range sets {
+		for i := int64(0); i < set.Count; i++ {
+			podSpec := set.PodSpec
+			nodeStickiness := podSpec.NodeStickiness
+			milliCPU := podSpec.MilliCPU
+			memory := podSpec.Memory
+			req := apiv1.ResourceList{}
+			req[apiv1.ResourceCPU] = *resource.NewMilliQuantity(milliCPU, resource.DecimalSI)
+			req[apiv1.ResourceMemory] = *resource.NewQuantity(memory, resource.DecimalSI)
+			var name string
+			if set.Count > 1 {
+				name = fmt.Sprintf("%s-%d", set.Name, i)
+			} else {
+				name = set.Name
+			}
+			pod := &apiv1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              fmt.Sprintf("placeholder-%s-%dm-%dMB", name, milliCPU, memory/1024/1024),
```

Comment:
> This should start with the name of replicaset. And the name of replicaset should contain cluster-autoscaler and slack (the pods can be written to logs).

Agreed. I will address that in an upcoming commit.

> what about events - do they work with fake pods?

Which kind of events do you mean? Could you provide me examples so that I can take extra care on them?